From bdafd42057ecafe263c350b887debb7d28959c27 Mon Sep 17 00:00:00 2001
From: Glen Yeldho <glen.yeldho@arm.com>
Date: Mon, 8 Apr 2024 23:56:39 +0530
Subject: [PATCH] feat(neoverse_rd): adds mapping in SRAM Region for RAS Einj

This patch adds mapping for RAS Einj buffer of Non-secure type in SRAM region.
RAS Einj buffer will be used by EDK2 and Kernel to perform error
injection operation using Linux sysfs interfaces.

Signed-off-by: Glen Yeldho <glen.yeldho@arm.com>
Change-Id: I869973794d4aaaee43f9fb0a5b964892bcfce3dd

Signed-off-by: Jon Mason <jon.mason@arm.com>
Upstream-Status: Pending
---
 .../common/include/nrd2/nrd_plat_arm_def2.h    | 18 ++++++++++++------
 plat/arm/board/neoverse_rd/common/nrd_plat2.c  |  2 ++
 2 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/plat/arm/board/neoverse_rd/common/include/nrd2/nrd_plat_arm_def2.h b/plat/arm/board/neoverse_rd/common/include/nrd2/nrd_plat_arm_def2.h
index aa0a486773cc..058ba7866e9c 100644
--- a/plat/arm/board/neoverse_rd/common/include/nrd2/nrd_plat_arm_def2.h
+++ b/plat/arm/board/neoverse_rd/common/include/nrd2/nrd_plat_arm_def2.h
@@ -60,13 +60,13 @@
 
 #if defined(IMAGE_BL31)
 # if SPM_MM || (SPMC_AT_EL3 && SPMC_AT_EL3_SEL0_SP)
-#  define PLAT_ARM_MMAP_ENTRIES		(10  + ((NRD_CHIP_COUNT - 1) * 3))
-#  define MAX_XLAT_TABLES		(8  + ((NRD_CHIP_COUNT - 1) * 3))
-#  define PLAT_SP_IMAGE_MMAP_REGIONS	U(12)
-#  define PLAT_SP_IMAGE_MAX_XLAT_TABLES	U(14)
+#  define PLAT_ARM_MMAP_ENTRIES		(12  + ((NRD_CHIP_COUNT - 1) * 3))
+#  define MAX_XLAT_TABLES		(10  + ((NRD_CHIP_COUNT - 1) * 3))
+#  define PLAT_SP_IMAGE_MMAP_REGIONS	14
+#  define PLAT_SP_IMAGE_MAX_XLAT_TABLES	16
 # else
-#  define PLAT_ARM_MMAP_ENTRIES		(5 + ((NRD_CHIP_COUNT - 1) * 3))
-#  define MAX_XLAT_TABLES		(6 + ((NRD_CHIP_COUNT - 1) * 3))
+#  define PLAT_ARM_MMAP_ENTRIES		(7 + ((NRD_CHIP_COUNT - 1) * 3))
+#  define MAX_XLAT_TABLES		(9 + ((NRD_CHIP_COUNT - 1) * 3))
 # endif
 #elif defined(IMAGE_BL32)
 # define PLAT_ARM_MMAP_ENTRIES		U(8)
@@ -204,6 +204,12 @@ ENABLE_FEAT_RAS && FFH_SUPPORT
 
 #define PLAT_ARM_NSRAM_BASE		NRD_CSS_NS_SRAM_BASE
 #define PLAT_ARM_NSRAM_SIZE		NRD_CSS_NS_SRAM_SIZE
+
+#define ARM_MAP_NSRAM             MAP_REGION_FLAT(                     \
+                                               PLAT_ARM_NSRAM_BASE,    \
+                                               PLAT_ARM_NSRAM_SIZE,    \
+                                               MT_MEMORY | MT_RW | MT_NS)
+
 /*
  * Required platform porting definitions common to all ARM CSS SoCs
  */
diff --git a/plat/arm/board/neoverse_rd/common/nrd_plat2.c b/plat/arm/board/neoverse_rd/common/nrd_plat2.c
index 8f97cae0765f..3b7b9383acad 100644
--- a/plat/arm/board/neoverse_rd/common/nrd_plat2.c
+++ b/plat/arm/board/neoverse_rd/common/nrd_plat2.c
@@ -72,6 +72,7 @@ const mmap_region_t plat_arm_mmap[] = {
 #if IMAGE_BL31
 const mmap_region_t plat_arm_mmap[] = {
 	NRD_CSS_SHARED_RAM_MMAP(0),
+        ARM_MAP_NSRAM,
 #ifdef PLAT_ARM_MEM_PROT_ADDR
 	ARM_V2M_MAP_MEM_PROTECT,
 #endif
@@ -92,6 +93,7 @@ const mmap_region_t plat_arm_secure_partition_mmap[] = {
 	NRD_ROS_PLATFORM_PERIPH_USER_MMAP,
 	ARM_SP_IMAGE_MMAP,
 	ARM_SP_IMAGE_NS_BUF_MMAP,
+	ARM_MAP_NSRAM,
 #if ENABLE_FEAT_RAS && FFH_SUPPORT
 	NRD_CSS_SP_CPER_BUF_MMAP,
 #endif
